### PR TITLE
[MODFISTO-244] New restrictions for fund and expense class codes

### PR DESCRIPTION
**Pull request by damien-git**
## Purpose
[MODFISTO-244](https://issues.folio.org/browse/MODFISTO-244) - Add validation for special character to Fund Code and Expense class code and prepare migration script

## Approach
- Updated acq-models with the new restrictions
- Added a migration script to convert codes
- Incremented the API version for expense classes

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by AndreiMPV — APPROVED** [review]

File: src/main/resources/templates/db_scripts/schema.json [hunk 1]
```diff
@@ -80,6 +80,11 @@
       "run": "after",
       "snippetPath": "budget_encumbrances_rollover.sql",
       "fromModuleVersion": "mod-finance-storage-7.1.1"
+    },
+    {
+      "run": "after",
+      "snippetPath": "migration/restrictions_for_fund_and_ec_codes.sql",
+      "fromModuleVersion": "mod-finance-storage-7.2.0"
```
Comment (AndreiMPV) [inline src/main/resources/templates/db_scripts/schema.json:87]: Please change it to 8.0.0. As major version of the API was changed then major version of the module should be released

**Review by AndreiMPV — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -2,7 +2,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>org.folio</groupId>
   <artifactId>mod-finance-storage</artifactId>
-  <version>7.2.0-SNAPSHOT</version>
+  <version>8.0.0-SNAPSHOT</version>
```
Comment (grabsefx) [inline pom.xml:5]: Please, do not edit project version mannualy. This will be don in scope of release PR

File: NEWS.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-## 7.2.0 - Unreleased
+## 8.0.0 - Unreleased
```
Comment (grabsefx) [inline NEWS.md:1]: This change will be done in scope of release PR

**Review by AndreiMPV — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (damien-git) [reply pom.xml:5]: OK, but then how will we know we have to increase the major version when releasing ?

Comment (grabsefx) [reply pom.xml]: Person who will do the release should quickly review the stories and make a decision about release version.

**Pull request merged**